pkg/system/phase2_creating.go

```diff
@@ -152,6 +159,46 @@ func (r *Reconciler) ReconcilePhaseCreatingForMainClusters() error {
 	return nil
 }
 
+func (r *Reconciler) UpgradeDb() error {
+	var result controllerutil.OperationResult
+	var err error
+
+	// Check the need for upgrade
+	if r.NooBaa.Status.PostgresUpdatePhase == nbv1.UpgradePhasePrepare {
+		r.NooBaa.Status.PostgresUpdatePhase = nbv1.UpgradePhaseUpgrade
```

Comment:
in the prepare phase you should scale down the core and endpoints, so they will not run during the upgrade. after the DB is upgraded successfully you should restore it back to normal

Reply:
Sure. Added it in the updated PR